File: libraries/managed-exceptions-lib/src/python/managed_exceptions/authorization/unauthorized_exception.py
from http import HTTPStatus
from managed_exceptions.error_details import ErrorDetails
from managed_exceptions.managed_exception import ManagedException

class UnauthorizedException(ManagedException):
    def __init__(self, message: str, diagnostic_details: dict[str, str] = {}):
        super().__init__(ErrorDetails(
            status_code=HTTPStatus.FORBIDDEN,
            diagnostic_code="00403",
            diagnostic_details=diagnostic_details,
            message=message
        ))

File: services/Sphana/Libraries/Sphana.ManagedException/Authorization/UnauthorizedException.cs
using System.Net;

namespace Sphana.ManagedException.Authorization;

/// <summary>
/// Thrown when a request is authenticated but lacks sufficient permissions.
/// </summary>
public class UnauthorizedException : ManagedException
{
    public UnauthorizedException(string message, Dictionary<string, string>? diagnosticDetails = null)
        : base(HttpStatusCode.Forbidden, "UNAUTHORIZED", message, diagnosticDetails)
    {
    }
}